rfc/rfc006-distributed-registry.md

```diff
@@ -1,378 +1,582 @@
-   * Assert all proofs authenticate the same vendor.
-5. Validate **certs**:
-   * Assert there is at least one certificate.
-   * Assert all certificates conform to the vendor CA certificate specification (see [RFC008](rfc008-certificate-structure.md)).
-   * Assert all certificates have the same MUST contain the same vendor ID.
-
-If any of these steps fail the registration or update SHALL NOT be processed.
-
-#### Revoking CA certificates
-When a vendor loses or leaks its private key, the vendor SHOULD remove the certificate from its registration to avoid
-malicious parties exploiting it. However, other parties MUST regard signatures created before removal of the certificate
- as valid. How the vendor SHOULD change its CA certificate depends on the situation:
-
-##### CA certificate private key loss
-In case the vendor loses its CA private key it MUST register a new vendor CA certificate. This can be done in two ways,
-depending on what was lost:
-
-1. If the vendor still has a valid signing certificate it COULD sign the update using that private key.
-2. Otherwise, when the signing certificate has expired or the private key is lost as well, the vendor MUST also provide
-new proof to authenticate ownership of the vendor. The new proof MUST be the first operation in the **patch** field.
-
-Example update:
+The DID document has the following basic requirements:
+
+- the `id` MUST be generated according to the method specified at the beginning of §3
+- at least 1 key MUST be present in the `verificationMethod` and `authentication`
+- all key references in `authentication` MUST refer to keys listed under `verificationMethods`
+
+Each key listed in `verificationMethod` MUST have an `id` equal to the DID followed by a `#` and the public key fingerprint according to [rfc7638](https://tools.ietf.org/html/rfc7638)
+Each key MUST be of type `JsonWebKey2020` according to §5.3.1 of the [did-core-spec](https://www.w3.org/TR/did-core/#key-types-and-formats)
+
+The `controller` field MAY be present. This RFC follows the [did-core-spec](https://www.w3.org/TR/did-core/#did-controller).
+
+The `assertion` field MAY be present. 
+Keys referenced from this field are used for signing Verifiable Credentials/Presentations and for signing JWTs in the OAuth flow.
+
+Example DID document:
+
 ```json
 {
-  "patch": [
-    { "op": "add", "path": "/prfs", "value": "some new proof" },
-    { "op": "replace", "path": "/crts", "value": ["CA certificate as base64 ASN.1"] }
-  ]
+  "@context": [ "https://www.w3.org/ns/did/v1" ],
+  "id": "did:nuts:123",
+  "controller": [
+    "did:nuts:123",
+    "did:nuts:7368245"
+  ],
+  "verificationMethod": [{
+    "id": "did:nuts:123#_TKzHv2jFIyvdTGF1Dsgwngfdg3SH6TpDv0Ta1aOEkw",
+    "controller": "did:nuts:123",
+    "type": "JsonWebKey2020",
+    "publicKeyJwk": {
+      "crv": "P-256",
+      "x": "38M1FDts7Oea7urmseiugGW7tWc3mLpJh6rKe7xINZ8",
+      "y": "nDQW6XZ7b_u2Sy9slofYLlG03sOEoug3I0aAPQ0exs4",
```

Comment:
remove the y?

Reply:
See `crypto/ecdsa/ecdsa.go:60`, ECDSA public key contains both X and Y.

You might be confused with edDSA, where the second component ('d') is the private part